src/System.Text.Json/src/System/Text/Json/Node/JsonArrayEnumerator.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    ///   Supports an iteration over a JSON array.
+    /// </summary>
+    public struct JsonArrayEnumerator : IEnumerator<JsonNode>
+    {
+        private readonly IEnumerator<JsonNode> _enumerator;
```

Comment:
Should/can this be `private readonly List<JsonNode>.Enumerator _enumerator;`?

Reply:
I can change it. Why would this be better?

Reply:
List returns a struct-based enumerator.  This is a struct-based enumerator.  If you store `IEnumerator<JsonNode>` then you have to pay the GC cost of boxing and the runtime cost of interface virtual dispatch.  If the answer is that it's always the List struct-based enumerator then we can keep the optimization (and perhaps have to change it later).

Really it's that this whole type only makes sense if you're avoiding boxing the list enumerator; otherwise you'd just return it directly.